From bc9dad5cf7c63364f5fda0fc7cb816bb6eb8b664 Mon Sep 17 00:00:00 2001
From: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
Date: Fri, 23 May 2025 18:37:25 +0900
Subject: [PATCH] drivers: clk: r8a779g0-cpg-mssr: backport from BSP
 kernel

Upstream-Status: Inappropriate [no upstream]

Signed-off-by: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
---
 drivers/clk/renesas/r8a779g0-cpg-mssr.c | 64 +++++++++++++++++++++++++
 1 file changed, 64 insertions(+)

diff --git a/drivers/clk/renesas/r8a779g0-cpg-mssr.c b/drivers/clk/renesas/r8a779g0-cpg-mssr.c
index afb86973e771..74a2a755ead1 100644
--- a/drivers/clk/renesas/r8a779g0-cpg-mssr.c
+++ b/drivers/clk/renesas/r8a779g0-cpg-mssr.c
@@ -165,8 +165,27 @@ static const struct cpg_core_clk r8a779g0_core_clks[] __initconst = {
 
 static const struct mssr_mod_clk r8a779g0_mod_clks[] __initconst = {
 	DEF_MOD("rgx",		  0,	R8A779G0_CLK_ZG),
+	DEF_MOD("smpo0",	  5,	R8A779G0_CLK_SVD1_VIP),
+	DEF_MOD("smps0",	  7,	R8A779G0_CLK_SVD1_VIP),
+	DEF_MOD("umfl0",	  9,	R8A779G0_CLK_SVD1_VIP),
 	DEF_MOD("isp0",		 16,	R8A779G0_CLK_S0D2_VIO),
 	DEF_MOD("isp1",		 17,	R8A779G0_CLK_S0D2_VIO),
+	DEF_MOD("impcnn",	 22,	R8A779G0_CLK_S0D6_PER),
+	DEF_MOD("spmc",		 23,	R8A779G0_CLK_S0D6_PER),
+	DEF_MOD("imp0",		 24,	R8A779G0_CLK_S0D6_PER),
+	DEF_MOD("imp1",		 25,	R8A779G0_CLK_S0D6_PER),
+	DEF_MOD("impdmac0",	 26,	R8A779G0_CLK_SVD1_IR),
+	DEF_MOD("imppsc",	 27,	R8A779G0_CLK_SVD1_IR),
+
+	DEF_MOD("imp2",		103,	R8A779G0_CLK_S0D3),
+	DEF_MOD("imp3",		104,	R8A779G0_CLK_S0D3),
+	DEF_MOD("impdmac1",	105,	R8A779G0_CLK_S0D3),
+	DEF_MOD("impdta",	116,	R8A779G0_CLK_S0D3),
+	DEF_MOD("impslv",	118,	R8A779G0_CLK_S0D6_PER),
+	DEF_MOD("spmi",		120,	R8A779G0_CLK_S0D6_PER),
+	DEF_MOD("adg",		122,	R8A779G0_CLK_S0D6_PER),
+	DEF_MOD("advfs",	123,	R8A779G0_CLK_S0D6_PER),
+
 	DEF_MOD("avb0",		211,	R8A779G0_CLK_S0D4_HSC),
 	DEF_MOD("avb1",		212,	R8A779G0_CLK_S0D4_HSC),
 	DEF_MOD("avb2",		213,	R8A779G0_CLK_S0D4_HSC),
@@ -176,6 +195,7 @@ static const struct mssr_mod_clk r8a779g0_mod_clks[] __initconst = {
 	DEF_MOD("dis0",		411,	R8A779G0_CLK_VIOBUSD2),
 	DEF_MOD("dsitxlink0",	415,	R8A779G0_CLK_VIOBUSD2),
 	DEF_MOD("dsitxlink1",	416,	R8A779G0_CLK_VIOBUSD2),
+	DEF_MOD("fcpcs",	507,	R8A779G0_CLK_S0D3),
 	DEF_MOD("fcpvd0",	508,	R8A779G0_CLK_VIOBUSD2),
 	DEF_MOD("fcpvd1",	509,	R8A779G0_CLK_VIOBUSD2),
 	DEF_MOD("hscif0",	514,	R8A779G0_CLK_SASYNCPERD1),
@@ -188,9 +208,16 @@ static const struct mssr_mod_clk r8a779g0_mod_clks[] __initconst = {
 	DEF_MOD("i2c3",		521,	R8A779G0_CLK_S0D6_PER),
 	DEF_MOD("i2c4",		522,	R8A779G0_CLK_S0D6_PER),
 	DEF_MOD("i2c5",		523,	R8A779G0_CLK_S0D6_PER),
+	DEF_MOD("imr0",			525,	R8A779G0_CLK_VCBUSD2),
+	DEF_MOD("imr1",			526,	R8A779G0_CLK_VCBUSD2),
+	DEF_MOD("imr2",			527,	R8A779G0_CLK_VCBUSD2),
+	DEF_MOD("ims0",			529,	R8A779G0_CLK_VCBUSD2),
+	DEF_MOD("ims1",			530,	R8A779G0_CLK_VCBUSD2),
+
 	DEF_MOD("irqc",		611,	R8A779G0_CLK_CL16M),
 	DEF_MOD("ispcs0",	612,	R8A779G0_CLK_S0D2_VIO),
 	DEF_MOD("ispcs1",	613,	R8A779G0_CLK_S0D2_VIO),
+	DEF_MOD("ivcp1e",	616,	R8A779G0_CLK_VCBUS),
 	DEF_MOD("msi0",		618,	R8A779G0_CLK_MSO),
 	DEF_MOD("msi1",		619,	R8A779G0_CLK_MSO),
 	DEF_MOD("msi2",		620,	R8A779G0_CLK_MSO),
@@ -201,6 +228,11 @@ static const struct mssr_mod_clk r8a779g0_mod_clks[] __initconst = {
 	DEF_MOD("pciec1",	625,	R8A779G0_CLK_S0D2_HSC),
 	DEF_MOD("pwm",		628,	R8A779G0_CLK_SASYNCPERD4),
 	DEF_MOD("rpc-if",	629,	R8A779G0_CLK_RPCD2),
+	DEF_MOD("rtdm0",	630,	R8A779G0_CLK_S0D4_RT),
+	DEF_MOD("rtdm1",	631,	R8A779G0_CLK_S0D4_RT),
+
+	DEF_MOD("rtdm2",	700,	R8A779G0_CLK_S0D4_RT),
+	DEF_MOD("rtdm3",	701,	R8A779G0_CLK_S0D4_RT),
 	DEF_MOD("scif0",	702,	R8A779G0_CLK_SASYNCPERD4),
 	DEF_MOD("scif1",	703,	R8A779G0_CLK_SASYNCPERD4),
 	DEF_MOD("scif3",	704,	R8A779G0_CLK_SASYNCPERD4),
@@ -214,8 +246,19 @@ static const struct mssr_mod_clk r8a779g0_mod_clks[] __initconst = {
 	DEF_MOD("tmu3",		716,	R8A779G0_CLK_SASYNCPERD2),
 	DEF_MOD("tmu4",		717,	R8A779G0_CLK_SASYNCPERD2),
 	DEF_MOD("tpu0",		718,	R8A779G0_CLK_SASYNCPERD4),
+	DEF_MOD("caiplite_wrapper0",    720,    R8A779G0_CLK_S0D2),
+	DEF_MOD("caiplite0",            721,    R8A779G0_CLK_S0D1_VIO),
+	DEF_MOD("caiplite1",            722,    R8A779G0_CLK_S0D1_VIO),
+	DEF_MOD("caiplite2",            723,    R8A779G0_CLK_S0D1_VIO),
+	DEF_MOD("caiplite3",            724,    R8A779G0_CLK_S0D1_VIO),
+	DEF_MOD("caiplite4",            725,    R8A779G0_CLK_S0D1_VIO),
+	DEF_MOD("caiplite5",            726,    R8A779G0_CLK_S0D1_VIO),
+	DEF_MOD("caiplite6",            727,    R8A779G0_CLK_S0D1_VIO),
+	DEF_MOD("caiplite7",            728,    R8A779G0_CLK_S0D1_VIO),
+	DEF_MOD("vcp4l",		729,	R8A779G0_CLK_S0D6_RT),
 	DEF_MOD("vin00",	730,	R8A779G0_CLK_S0D4_VIO),
 	DEF_MOD("vin01",	731,	R8A779G0_CLK_S0D4_VIO),
+
 	DEF_MOD("vin02",	800,	R8A779G0_CLK_S0D4_VIO),
 	DEF_MOD("vin03",	801,	R8A779G0_CLK_S0D4_VIO),
 	DEF_MOD("vin04",	802,	R8A779G0_CLK_S0D4_VIO),
@@ -232,6 +275,11 @@ static const struct mssr_mod_clk r8a779g0_mod_clks[] __initconst = {
 	DEF_MOD("vin17",	813,	R8A779G0_CLK_S0D4_VIO),
 	DEF_MOD("vspd0",	830,	R8A779G0_CLK_VIOBUSD2),
 	DEF_MOD("vspd1",	831,	R8A779G0_CLK_VIOBUSD2),
+
+	DEF_MOD("wcrc0",		903,	R8A779G0_CLK_S0D2_RT),
+	DEF_MOD("wcrc1",		904,	R8A779G0_CLK_S0D2_RT),
+	DEF_MOD("wcrc2",		905,	R8A779G0_CLK_S0D2_RT),
+	DEF_MOD("wcrc3",		906,	R8A779G0_CLK_S0D2_RT),
 	DEF_MOD("wdt1:wdt0",	907,	R8A779G0_CLK_R),
 	DEF_MOD("cmt0",		910,	R8A779G0_CLK_R),
 	DEF_MOD("cmt1",		911,	R8A779G0_CLK_R),
@@ -246,8 +294,24 @@ static const struct mssr_mod_clk r8a779g0_mod_clks[] __initconst = {
 	DEF_MOD("vspx1",	1029,	R8A779G0_CLK_S0D1_VIO),
 	DEF_MOD("fcpvx0",	1100,	R8A779G0_CLK_S0D1_VIO),
 	DEF_MOD("fcpvx1",	1101,	R8A779G0_CLK_S0D1_VIO),
+	DEF_MOD("cve0",		2706,	R8A779G0_CLK_SVD1_IR),
+	DEF_MOD("cve1",		2707,	R8A779G0_CLK_SVD1_IR),
+	DEF_MOD("cve2",		2708,	R8A779G0_CLK_SVD1_IR),
+	DEF_MOD("cve3",		2709,	R8A779G0_CLK_SVD1_IR),
+	DEF_MOD("impsdmac0",	2712,	R8A779G0_CLK_SVD1_VIP),
+	DEF_MOD("impsdmac1",	2713,	R8A779G0_CLK_SVD1_VIP),
 	DEF_MOD("tsn",		2723,	R8A779G0_CLK_S0D4_HSC),
+	DEF_MOD("vdsp0_bus",	2801,	R8A779G0_CLK_SVD1_IR),
+	DEF_MOD("vdsp1_bus",	2802,	R8A779G0_CLK_SVD1_IR),
+	DEF_MOD("vdsp2_bus",	2803,	R8A779G0_CLK_SVD1_IR),
+	DEF_MOD("vdsp3_bus",	2804,	R8A779G0_CLK_SVD1_IR),
+	DEF_MOD("paptop",	2806,	R8A779G0_CLK_S0D6_PER),
+	DEF_MOD("papsdma",	2807,	R8A779G0_CLK_S0D6_PER),
 	DEF_MOD("dsc",		2819,	R8A779G0_CLK_VIOBUSD2),
+	DEF_MOD("vdsp0_csb",	2821,	R8A779G0_CLK_SVD1_IR),
+	DEF_MOD("vdsp1_csb",	2830,	R8A779G0_CLK_SVD1_IR),
+	DEF_MOD("vdsp2_csb",	2907,	R8A779G0_CLK_SVD1_IR),
+	DEF_MOD("vdsp3_csb",	2916,	R8A779G0_CLK_SVD1_IR),
 	DEF_MOD("ssiu",		2926,	R8A779G0_CLK_S0D6_PER),
 	DEF_MOD("ssi",		2927,	R8A779G0_CLK_S0D6_PER),
 };
-- 
2.34.1

