Upstream-Status: Inappropriate [oe specific]

---

diff --git a/libweston/meson.build b/libweston/meson.build
index 7c2fc12..8abd1cb 100644
--- a/libweston/meson.build
+++ b/libweston/meson.build
@@ -99,7 +99,7 @@ srcs_libweston = [
 subdir('desktop')
 subdir('shell-utils')
 
-dep_gbm = dependency('gbm', required: false, version: '>= 21.1.1')
+dep_gbm = dependency('gbm', required: false)
 if dep_gbm.found()
 	config_h.set('HAVE_GBM', '1')
 endif
